File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isBalanced(TreeNode* root) {
        if (!root) return true;
        int first= getHeight( root->left);
        int second = getHeight(root->right);
        if(abs(first - second) > 1 ) return false;
        return isBalanced(root->left) && isBalanced(root->right);
    }
    int getHeight(TreeNode* curr){
        if(curr== nullptr) return 0;
        return max(getHeight(curr->left), getHeight(curr->right))+1;
    }
};

File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1: return nums[0]
        def dp_algo (nums):
            if not nums: return 0
            dp= {}
            for i in range(len (nums)):
                if i == 0:
                    dp[0]= nums[0]
                elif i == 1:
                    dp[1]= max(nums[0], nums[1])
                else:
                    dp[i]= max(dp[i-1], nums[i]+dp[i-2])
            return dp[len(nums)-1]
        first= dp_algo(nums[:-1])
        second=dp_algo(nums[1:])
        return max(first,second)